src/Forum/Forum.service.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  Collection,
+  CollectionAttributes,
+  ThirdPartyCollectionAttributes,
+} from '../Collection'
+import { FullItem } from '../Item'
+import { createPost, getPost, updatePost } from './client'
+import {
+  buildCollectionForumPost,
+  buildCollectionForumUpdateReply,
+} from './utils'
+
+export class ForumService {
+  async upsertThirdPartyCollectionForumPost(
+    collection: ThirdPartyCollectionAttributes,
+    items: FullItem[]
+  ): Promise<string | undefined> {
+    if (collection.forum_id) {
+      const postData = await getPost(collection.forum_id)
+      await updatePost(
+        collection.forum_id,
+        buildCollectionForumUpdateReply(postData.raw, items)
+      )
+      return
```

Comment:
wdyt about returning the link here too?

Reply:
Is it possible to get it with the data that I have here or should I modify the `updatePost` so it returns the same as `createPost` and we can get the link there?

Reply:
I like how you updated the `updatePost` method to return the same link. sorry for the late reply